yarn/datadog_checks/yarn/yarn.py

```diff
@@ -245,16 +246,38 @@ def _yarn_app_metrics(self, rm_address, app_tags, addl_tags):
                 )
 
     def _get_app_tags(self, app_json, app_tags):
+        split_app_tags = self.instance.get('split_yarn_application_tags', DEFAULT_SPLIT_YARN_APPLICATION_TAGS)
         tags = []
         for dd_tag, yarn_key in iteritems(app_tags):
             try:
                 val = app_json[yarn_key]
                 if val:
-                    tags.append('{tag}:{value}'.format(tag=dd_tag, value=val))
+                    if split_app_tags and yarn_key == 'applicationTags':
+                        splitted_tags = self.__split_yarn_application_tags(val, dd_tag)
+                        tags.extend(splitted_tags)
+                    else:
+                        tags.append('{tag}:{value}'.format(tag=dd_tag, value=val))
             except KeyError:
                 self.log.error("Invalid value %s for application_tag", yarn_key)
         return tags
 
+    def __split_yarn_application_tags(self, application_tags, dd_tag):
```

Comment:
As a point of style, we usually avoid private methods with double underscores.  Changing this to a single underscore would work though.

Reply:
done